### DISCUSSION TITLE
Multiple users connect to a single codespace

### DISCUSSION THREAD

**Discussion by brianmhunt** [community/community#8139] [Codespaces]
One of the benefits we are looking for out of Codespaces is for developers to share their work with each other and with other teams.  However, it seems like only one user can be connected to a Codespace.

The sharing can be broken down into a few separate cases:

1. connecting to the Codespace of another user in VS Code
2. connecting to the Codespace of another user in the browser
3. connecting to the shared ports of a Codespace at `https://{user}-{project}-{repo}-{id}-{port}`

For items 2 and 3 it seems anytime a user other than the authenticated user attempts to access the URL they get a 404.

Of these, the most important for our use case is the third (sharing ports) among Github users because it would allow sharing of work in an interim development state, so we can get team feedback and eliminate the need for a deploy to another container.

**Comment by brianmhunt** [comment]
I came across the new API in the [Blog post: "New Codespaces features launching at Universe 2021"](https://github.blog/changelog/2021-10-27-new-codespaces-features-launching-at-universe-2021/):

> We also know that security and privacy are critical, and we've gotten a lot of feedback on providing additional visibility options for forwarded ports beyond public and private. Today we're launching a third option: org visible ports, which are accessible to any user in the organization the codespace has been created in. This is great for securely collaborating with your teammates on new and exciting features in your codespaces.

```
gh codespace ports visibility 80:org -c monalisa-monalisa-octocat-1337h4x0r
```

... so it seems this is well on its way.

**Comment by shellerci** [comment]
Have you checked out the [Live Share Extension](https://visualstudio.microsoft.com/services/live-share/) for VSCode? It handles all the requirements you've listed pretty seamlessly. 

My only minor quibble is that the host of the codeshare session needs to manually launch a zsh session for a guest if they want shell access, but it's a two click operation and I totally understand why you might not want that to be an automated process.
